service/controller.go

```diff
@@ -361,9 +374,47 @@ func (s *service) CreateVolume(
 			return nil, status.Errorf(codes.Unknown, "Create Volume %s failed with error: %v", volName, err)
 		}
 
+		// set quota limits, if specified in NFS storage class
+		isQuotaEnabled := s.opts.IsQuotaEnabled
+		if isQuotaEnabled {
+			// get filesystem (NFS volume), newly created
+			fs, err := system.GetFileSystemByIDName(fsResp.ID, "")
+			if err != nil {
+				Log.Debugf("Find Volume response error: %v", err)
+				return nil, status.Errorf(codes.Unknown, "Find Volume response error: %v", err)
+			}
+			path, ok := params[KeyPath]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeyPath)
+			}
+
+			softLimit, ok := params[KeySoftLimit]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeySoftLimit)
+			}
+
+			gracePeriod, ok := params[KeyGracePeriod]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeyGracePeriod)
+			}
+
+			// create quota for the filesystem
+			quotaID, err := s.createQuota(fsResp.ID, path, softLimit, gracePeriod, int(size), isQuotaEnabled, systemID)
+			if err != nil {
+				// roll back, delete the newly created volume
+				if err = system.DeleteFileSystem(fs.Name); err != nil {
+					return nil, status.Errorf(codes.Internal,
+						"rollback (deleting volume '%s') failed with error : '%v'", fs.Name, err.Error())
+				}
+				return nil, fmt.Errorf("error creating quota ('%s', '%d' bytes), abort, also successfully rolled back by deleting the newly created volume", fs.Name, size)
+			}
+			Log.Infof("Tree quota set for: %d bytes on directory: '%s', quota ID: %s", size, path, quotaID)
+		}
+
 		newFs, err := system.GetFileSystemByIDName(fsResp.ID, "")
 		if err != nil {
-			Log.Debugf("Find Volume response: %v Error: %v", newFs, err)
+			Log.Debugf("Find Volume response error: %v", err)
+			return nil, status.Errorf(codes.Unknown, "Find Volume response error: %v", err)
```

Comment:
same here

Reply:
Keeping it same to maintain consistency.